### DISCUSSION TITLE
Is it possible to checkout code before starting a service?

### DISCUSSION THREAD

**Discussion by ryanto** [community/community#25792] [Actions]
<p>I’m using the <a href="https://docs.github.com/en/actions/using-containerized-services/about-service-containers" rel="noopener nofollow ugc">service containers</a> to spin up a couple of services my tests need. It’s working great!</p>
<p>However, I’m running into an issue. One of the services I’m using expects to have a configuration file (exposed by a volume) before it starts. This configuration file lives in my repo, but it looks the service start before the checkout action runs.</p>
<p>Is it possible to have checkout happen before the services start? Any tips for making a user supplied config is available to a service container.</p>
<p>Thank you!</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="ryanto" data-post="1" data-topic="216863">
<div class="title">
<div class="quote-controls"></div>
 ryanto:</div>
<blockquote>
<p>Is it possible to have checkout happen before the services start? Any tips for making a user supplied config is available to a service container.</p>
</blockquote>
</aside>
<p>You’ll probably have to start your container(s) manually to achieve that, e.g. with <code>docker run</code> commands in the workflow after checkout. Just make sure to attach them to the Docker network with the name set in <code>job.container.network</code> (see <a href="https://docs.github.com/en/actions/learn-github-actions/contexts#job-context"><code>job</code> context</a>) and they should be reachable the same as the services configured for the job.</p>
